kyuubi-main/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala

```diff
@@ -71,12 +73,24 @@ class ExecuteStatement(
 
       val req = new TExecuteStatementReq(remoteSessionHandle, statement)
       req.setRunAsync(shouldRunAsync)
+      req.setQueryTimeout(getQueryTimeout)
       val resp = client.ExecuteStatement(req)
       verifyTStatus(resp.getStatus)
       _remoteOpHandle = resp.getOperationHandle
     } catch onError()
   }
 
+  private def getQueryTimeout: Long = {
```

Comment:
move this to sessionManager?

Reply:
We only do the timeout check only in `ExecteStatement` operation, no stronge opinion to move that..